platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/main/java/com/swirlds/demo/consistency/ConsistencyTestingToolRound.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param round        the round to convert
+     * @param currentState the long state value of the application after the round has been applied
+     * @return the input round, converted to a {@link ConsistencyTestingToolRound}
+     */
+    @NonNull
+    public static ConsistencyTestingToolRound fromRound(final @NonNull Round round, final long currentState) {
+        Objects.requireNonNull(round);
+
+        final List<Long> transactionContents = new ArrayList<>();
+
+        round.forEachTransaction(transaction -> transactionContents.add(byteArrayToLong(transaction.getContents(), 0)));
+
+        return new ConsistencyTestingToolRound(round.getRoundNum(), currentState, transactionContents);
+    }
+
+    /**
+     * Construct a {@link ConsistencyTestingToolRound} from a string representation
+     *
+     * @param roundString the string representation of the round
+     * @return the new {@link ConsistencyTestingToolRound}, or null if parsing failed
+     */
+    @Nullable
+    public static ConsistencyTestingToolRound fromString(final @NonNull String roundString) {
+        Objects.requireNonNull(roundString);
+
+        try {
+            final List<String> fields =
+                    Arrays.stream(roundString.split(FIELD_SEPARATOR)).toList();
+
+            String field = fields.get(0);
+            final long roundNumber = Long.parseLong(field.substring(ROUND_NUMBER_STRING.length()));
+
+            field = fields.get(1);
+            final long currentState = Long.parseLong(field.substring(CURRENT_STATE_STRING.length()));
+
+            field = fields.get(2);
+            final String transactionsString = field.substring(field.indexOf("[") + 1, field.indexOf("]"));
+            final List<Long> transactionsContents = Arrays.stream(transactionsString.split(LIST_ELEMENT_SEPARATOR))
+                    .map(Long::parseLong)
+                    .toList();
+
+            return new ConsistencyTestingToolRound(roundNumber, currentState, transactionsContents);
+        } catch (final Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int compareTo(@NonNull ConsistencyTestingToolRound other) {
+        Objects.requireNonNull(other);
+
+        return Long.compare(this.roundNumber, other.roundNumber);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(final @Nullable Object other) {
+        if (other == null) {
+            return false;
+        }
+
+        if (other == this) {
+            return true;
+        }
+
+        if (other instanceof final ConsistencyTestingToolRound otherRound) {
+            return roundNumber == otherRound.roundNumber
+                    && currentState == otherRound.currentState
+                    && transactionsContents.equals(otherRound.transactionsContents);
+        }
+
+        return false;
+    }
+
+    @Override
+    @NonNull
+    public String toString() {
```

Comment:
Add a javadoc that says that the string produced by this method can be deserialized with `fromString()`. That way it's obvious to somebody in the future not to be careful when modifying this method.